Sources/Kafka/KafkaProducerEvent.swift

```diff
@@ -23,8 +23,6 @@ public enum KafkaProducerEvent: Sendable, Hashable {
         switch event {
         case .deliveryReport(results: let results):
```

Comment:
Hm.. I guess it is out of scope here but should we have backpressure for delivery reports as well?

Reply:
I don't think we can since they happen on the `eventsQueue` unless there is a way to separate events to different queues.

Reply:
Yes, the problem is that we should call `poll()` at regular intervals to serve any queued callbacks/events. As @FranzBusch points out, we would need `librdkafka` to have a separate "delivery reports queue" for backpressure to make sense here since some events like log events should still be served even when our `events` `AsyncSequence` with all the `.deliveryReport` events is suspended.

Reply:
Ah, yes, sure... We can only separate log queue from others...
So, we only can make some "unfair" backpressue, i.e. sometimes poll even if receive `stopProducing` which is dirty solution...

Reply:
Yes, we could separate the log queue but then there is also other events like commit confirmation

Reply:
Yeah, I got it, we may close this discussion, probably to think about it with separate case as it is out of scope for this PR anyway and might be not a problem (at least far).